public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ConfigEditor.tsx

```diff
@@ -35,78 +29,56 @@ export class ConfigEditor extends PureComponent<Props, State> {
     this.backendSrv = getBackendSrv();
     this.templateSrv = new TemplateSrv();
 
-    if (options.id) {
-      this.state.config.url = '/api/datasources/proxy/' + options.id;
-      this.init();
+    if (this.props.options.id) {
+      this.props.options.url = '/api/datasources/proxy/' + this.props.options.id;
     }
 
-    this.updateDatasource(this.state.config);
+    this.onOptionsUpdate(this.props.options);
   }
 
-  static getDerivedStateFromProps(props: Props, state: State) {
-    return {
-      ...state,
-      config: ConfigEditor.defaults(props.options),
-    };
-  }
-
-  static defaults = (options: any) => {
-    options.jsonData.cloudName = options.jsonData.cloudName || 'azuremonitor';
-
-    if (!options.jsonData.hasOwnProperty('azureLogAnalyticsSameAs')) {
-      options.jsonData.azureLogAnalyticsSameAs = true;
-    }
-
-    if (!options.hasOwnProperty('secureJsonData')) {
-      options.secureJsonData = {};
-    }
-
-    if (!options.hasOwnProperty('secureJsonFields')) {
-      options.secureJsonFields = {
-        clientSecret: false,
-        logAnalyticsClientSecret: false,
-        appInsightsApiKey: false,
-      };
-    }
-
-    return options;
-  };
-
   backendSrv: BackendSrv = null;
   templateSrv: TemplateSrv = null;
 
-  init = async () => {
+  async componentDidMount() {
     await this.getSubscriptions();
 
-    if (!this.state.config.jsonData.azureLogAnalyticsSameAs) {
+    if (!this.props.options.jsonData.azureLogAnalyticsSameAs) {
       await this.getLogAnalyticsSubscriptions();
     }
```

Comment:
Personal pref: I would keep data fetching outside of the component and store this in Redux. Simplifies testing of component and data fetching. Also what happens here if the user navigates away while these Promises are running?